sdl_android/src/main/java/com/smartdevicelink/api/BaseSubManager.java

```diff
@@ -0,0 +1,48 @@
+package com.smartdevicelink.api;
+
+import android.support.annotation.NonNull;
+
+import com.smartdevicelink.proxy.interfaces.ISdl;
+
+/**
+ * <strong>BaseSubManager</strong> <br>
+ *
+ * Note: This class is extended by SubManagers <br>
+ *
+ * It is broken down to these areas: <br>
+ *
+ * 1. <br>
+ */
+public abstract class BaseSubManager {
+
+	// states - if this gets more complicated we can move elsewhere
+	public enum ManagerState {
```

Comment:
For the possibility to expand this based on manager, maybe these should be constants instead of eunms. Just give them large values to separate them like `SETTING_UP = 0x00`, `READY = 0x30`, `SHUTDOWN = 0x60`